Source/DafnyLanguageServer/Language/SyntaxTreeVisitor.cs

```diff
@@ -505,9 +530,49 @@ public abstract class SyntaxTreeVisitor {
     public virtual void Visit(DisplayExpression displayExpression) {
     }
 
-    public virtual void Visit(ComprehensionExpr comprehensionExpression) {
+    private void VisitBoundVars(IBoundVarsBearingExpression boundVariablesBearingExpression) {
+      foreach (var boundVariable in boundVariablesBearingExpression.AllBoundVars) {
+        Visit(boundVariable);
+      }
+    }
+
+    private void VisitComprehensionBoundVarsAttributesRange(ComprehensionExpr comprehensionExpr) {
+      VisitBoundVars(comprehensionExpr);
+      VisitNullableAttributes(comprehensionExpr.Attributes);
+      VisitNullableExpression(comprehensionExpr.Range);
+    }
+
+    public virtual void Visit(LambdaExpr lambdaExpression) {
+      VisitComprehensionBoundVarsAttributesRange(lambdaExpression);
+      VisitNullableExpression(lambdaExpression.Term);
+    }
+
+    public virtual void Visit(ForallExpr forallExpression) {
+      VisitComprehensionBoundVarsAttributesRange(forallExpression);
+      VisitNullableExpression(forallExpression.Term);
     }
 
+    public virtual void Visit(ExistsExpr existsExpression) {
+      VisitComprehensionBoundVarsAttributesRange(existsExpression);
+      VisitNullableExpression(existsExpression.Term);
+    }
+
+    public virtual void Visit(SetComprehension setComprehensionExpression) {
+      VisitComprehensionBoundVarsAttributesRange(setComprehensionExpression);
+      if (!setComprehensionExpression.TermIsImplicit) {
+        VisitNullableExpression(setComprehensionExpression.Term);
+      }
+    }
+
```

Comment:
Some unnecessary line breaks.
Maybe we could extend the code formatter/.editorconfig to look for these.